services/ymax-planner/src/logger.ts

```diff
@@ -0,0 +1,59 @@
+import { AsyncLocalStorage } from 'node:async_hooks';
+import type { Console } from 'node:console';
+
+type TraceContext = {
+  traceId?: string;
+};
+
+const traceStore = new AsyncLocalStorage<TraceContext>();
+
+const tracePrefix = () => {
+  const { traceId } = traceStore.getStore() || {};
+  return traceId ? `[${traceId}] ` : '';
+};
+
+const withPrefix = (args: any[]) => {
+  const prefix = tracePrefix();
+  return prefix ? [prefix, ...args] : args;
+};
+
+/**
+ * Run a function within a trace context so downstream async work can reuse the
+ * same traceId without needing to thread it through parameters.
+ */
+export const runWithTrace = async <T>(
+  traceId: string,
+  fn: () => Promise<T> | T,
+): Promise<T> => traceStore.run({ traceId }, fn);
+
+export const runWithFlowTrace = async <T>(
+  portfolioKey: string,
+  flowKey: string,
+  fn: () => Promise<T> | T,
+): Promise<T> => {
+  const traceId = `${portfolioKey}.${flowKey}`;
+  return runWithTrace(traceId, fn);
+};
```

Comment:
This seems like it belongs in engine.ts rather than an otherwise-generic logger.ts library module.

Reply:
Debatable. It's the "ymax-planner logger". If it were in some other package I'd definitely agree. I would approve moving it tho if you like